### ISSUE TITLE
Unable to perform `aws iam` operations with `acp` only `asp`

### ISSUE THREAD

**Issue by choffa** [ohmyzsh/ohmyzsh#10083]
### Describe the bug

Having an issue where I am unable to perform `iam` actions (such as `aws iam list-users`) when setting the profile with `acp` but I am able to perform them when setting the profile with `asp`.

When doing `aws iam list-users` with `acp` set profile, I get the following error:
```
An error occurred (InvalidClientTokenId) when calling the ListUsers operation: The security token included in the request is invalid
```

### Steps to reproduce

1. Enable plugin `aws`
2. `asp` to profile with `AdministratorAccess` managed role
3. Perform `iam` action such as `aws iam list-users` and confirm expected behavior
4. Clear `asp` profile and reset with `acp`
5. Rerun `iam` action. Get empty response with error message

### Expected behavior

I expect credentials to work the same across the two methods.

### OS / Distro

macOS 10.15.7

### Zsh version

5.7.1

### Terminal emulator

Terminal

### Additional context

I have tried refreshing the access token, with no change in behaviour.

**Comment by tokongs** [comment]
Have you set the ARN of the role to be assumed in the config file?

**Comment by choffa** [comment]
Sorry for the late reply. No i haven't, what config file are you referring to?

**Comment by robbyrussell** [comment]
Hey @choffa, sorry — five years, and @tokongs asked you a good question that nobody followed up on.

This is a difference between the two commands rather than a bug. `asp` only sets `AWS_PROFILE`, so the CLI keeps using your IAM user's own keys and `aws iam` calls work normally. `acp` calls STS and exports `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY` and `AWS_SESSION_TOKEN`, which override the profile — and without a `role_arn` configured it falls back to `get-session-token`, whose credentials are restricted for IAM operations.

Short version: `acp` is for profiles that have a role configured, which the README covers. If you only want to switch profile, `asp` is the right one.

Closing as a config question rather than a defect.

🤖 Co-authored with Claude Code
